### PR TITLE
Customer rider endpoints

### PR DESCRIPTION
**Pull request by Samuel-Adeyeye**
Please review for merging...

### CODE REVIEW

**Comment by Samuel-Adeyeye** [comment]
Rider Not Customer

File: apps/server/src/controllers/customerController.ts
Comment (Don-christo) [inline apps/server/src/controllers/customerController.ts:1]: Delete this file nau. We don't need this any longer. Do you not see that we already have a registerCustomer function in authCustomers.ts file? I just wanted to let you know that there's no need for this code. It's null and void. Delete it, please.

File: apps/server/src/controllers/riderController.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        // You can perform additional validation or saving to the database here if needed
+
+        res.status(201).json({ message: 'Rider registered successfully' });
+    } catch (error) {
+        console.error('Error registering rider:', error);
+        res.status(500).send('Internal server error');
+    }
+};
```
Comment (Don-christo) [inline apps/server/src/controllers/riderController.ts:21]: The same thing applies to this. Remove this function entirely. There's an existing function for that.

File: apps/server/src/controllers/riderController.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        res.status(200).json({ riders });
+    } catch (error) {
+        console.error('Error fetching rider details:', error);
+        res.status(500).send('Internal server error');
+    }
+};
+
+// Controller function for retrieving a single rider from the database by id
+export const getRiderById = async (req: Request, res: Response): Promise<void> => {
+    const riderId: string = req.params.riderId;
```
Comment (Don-christo) [inline apps/server/src/controllers/riderController.ts:37]: What you did is wrong. This should be `req.params.id`. Update accordingly. And why are you making use of Promise<void>? There's no need for that. Remove please.

File: apps/server/src/controllers/riderController.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        res.status(201).json({ message: 'Rider registered successfully' });
+    } catch (error) {
+        console.error('Error registering rider:', error);
+        res.status(500).send('Internal server error');
+    }
+};
+
+// Controller function for retrieving details of riders
+export const getRiderDetails = async (req: Request, res: Response): Promise<void> => {
```
Comment (Don-christo) [inline apps/server/src/controllers/riderController.ts:24]: Remove Promise<void>.

File: apps/server/src/controllers/riderController.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    try {
+        // Fetch all riders details from the database
+        const riders = await Ryder.findAll();
+        res.status(200).json({ riders });
+    } catch (error) {
+        console.error('Error fetching rider details:', error);
+        res.status(500).send('Internal server error');
```
Comment (Don-christo) [inline apps/server/src/controllers/riderController.ts:31]: make use of existing code in the code base. There's a reusable code for HTTP status codes in the constants folder. Reference the registerCustomer function I implemented.

File: apps/server/src/controllers/riderController.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        if (!rider) {
+            res.status(404).json({ message: 'Rider not found' });
+            return;
+        }
+
+        res.status(200).json({ rider });
+    } catch (error) {
+        console.error('Error fetching rider by id:', error);
+        res.status(500).send('Internal server error');
```
Comment (Don-christo) [inline apps/server/src/controllers/riderController.ts:53]: Same thing here.

File: apps/server/src/controllers/riderController.ts
Comment (Don-christo) [inline apps/server/src/controllers/riderController.ts:1]: Remove the registerRider function and input all other functions in the authRiders.ts file and make the necessary changes, and then delete this file.

File: apps/server/src/app.ts [hunk 6]
```diff
@@ -69,6 +72,23 @@ app.use(function (err: HttpError, req: Request, res: Response) {
...
+});
+
+// Define route for retrieving rider details
+app.get('/riders', async (req, res) => {
+  await getRiderDetails(req, res);
+});
+app.get('/rider:id', async (req, res) => {
+  await getRiderById(req, res);
+});
+
```
Comment (Don-christo) [inline apps/server/src/app.ts:91]: There's a routes folder for you to do all this. There's no need to implement this in the app.ts file. Remove this please and update the routes folder accordingly.

File: apps/server/src/app.ts [hunk 7]
```diff
@@ -69,6 +72,23 @@ app.use(function (err: HttpError, req: Request, res: Response) {
...
+app.post('/register/rider', async (req, res) => {
+  await registerRider(req, res);
+});
+
+// Define route for retrieving rider details
+app.get('/riders', async (req, res) => {
+  await getRiderDetails(req, res);
+});
+app.get('/rider:id', async (req, res) => {
```
Comment (Don-christo) [inline apps/server/src/app.ts:88]: you have a wrong url link. It should be `'/rider/:id'`

File: apps/server/src/app.ts [hunk 8]
```diff
@@ -69,6 +72,23 @@ app.use(function (err: HttpError, req: Request, res: Response) {
...
+  await getRiderById(req, res);
```
Comment (Don-christo) [inline apps/server/src/app.ts:89]: Why do you need to async await the function when routing when you've already done that in the function? this is wrong. Remove the async await, please.

**Review by Don-christo — CHANGES_REQUESTED** [review]

**Review by Don-christo — APPROVED** [review]
LGTM! Nice work done.

**Pull request merged**